src/org/exist/Indexer.java

```diff
@@ -305,6 +305,9 @@ public void endDTD() {
     @Override
     public void endDocument() {
         if (!validate) {
+            if(indexListener != null) {
+                indexListener.endDocument(transaction);
```

Comment:
I don't need to send `DocumentImpl` as `setDocument` will have been called on the Index before `startDocument` is called on the StreamListener. `startDocument` and `endDocument` do not indicate the start or end of a DOM Document Node, rather they indicate that start and end of indexing a document.